### PR TITLE
Dm 1788

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-1788

## Description - what does this code do?
Changes to homepage geolocation:
- Adds autocomplete to facility search.
- Increased prominence of filter button.
- Adds unsuccessful adoption count to tooltip.
- Adds unsuccessful adoption entry to adoption list within the marker modal.

## Testing done - how did you test it/steps on how can another person can test it 
**Autocomplete:**
1. Browse to `/`.
2. Click on the `Filter results` button.
3. Scroll down to the input labelled `Facility name`.
4. Type in at least two letters and autocomplete should be triggered.

**Adoptions:**
1. Log in as an admin.
2. Browse to adoptions page of the practice editor for any practice.
3. Add an unsuccessful adoption entry.
4. Browse to `/`.
5. Find the facility that corresponds with the adoption entry from above and click on the marker. 
6. Make sure the unsuccessful adoption count increased by one.
7. Click `View more`.
8. Make sure the unsuccessful adoption was added to the list of items and it has the verbiage "abandoned."

## Screenshots, Gifs, Videos from application (if applicable)
Screen Shot 2020-06-04 at 4 40 01 PM
Screen Shot 2020-06-04 at 4 39 20 PM
Screen Shot 2020-06-04 at 4 40 21 PM

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/DDUSxYiD1MaWNrGFmMjY8d8a/Diffusion-Marketplace-Designs?node-id=1745%3A27776

## Acceptance criteria
- [Have "Unsuccessful" practices show up]
- [Need copy for how to phrase “Unsuccessful” date]
- [“Abandoned Date?”]
- [Facility autocomplete]
- [Make “Filters” button more prominent]
- [Tool tip Unsuccessful]

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/views/maps/_facility_auto_complete.js.erb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        $.each(strs, function(i, str) {
+            if (substrRegex.test(str)) {
+                matches.push(str);
+            }
+        });
+
+        cb(matches);
+    };
+};
+var facilityNameArr = []
```
Comment (ahay-agile6) [inline app/views/maps/_facility_auto_complete.js.erb:24]: let's not forget our semi-colons :slightly_smiling_face: ;

File: app/views/maps/_facility_auto_complete.js.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+$.each(facilities, function(key, value) {
+    facilityNameArr.push(value.OfficialStationName)
```
Comment (ahay-agile6) [inline app/views/maps/_facility_auto_complete.js.erb:26]: is there a way we can match what @emyl3 is doing with the Common Name in parenthesis?

File: app/views/maps/_home_map_filters.html.erb [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
+    <%= render partial: 'maps/facility_auto_complete', formats: [:js] %>
+  <% end %>
+<% end %>
+
 <% diffused_practices = @diffused_practices.map(&:practice).uniq.sort_by(&:name) %>
 <div id="filterResults" class="map-filters padding-y-0 position-absolute display-none">
   <div class="padding-left-1">
     <div class="margin-top-2 margin-bottom-105">
-      <span id="filterClose" class="close text-base-darkest" tabindex="0" role="button" aria-label="Close map filters"><span class="fas fa-times"></span></span>
+      <!--      <span id="filterClose" class="close text-base-darkest" tabindex="0" role="button" aria-label="Close map filters"><span class="fas fa-times"></span></span>-->
```
Comment (ahay-agile6) [inline app/views/maps/_home_map_filters.html.erb:11]: if we don't need this we can delete?

File: app/views/maps/_facility_auto_complete.js.erb (hunk 2 above)
Comment (emyl3) [reply app/views/maps/_facility_auto_complete.js.erb:26]: maybe I can push what I have rn and open up a WIP PR so you can see what I'm doing @JoshingYou1

Comment (emyl3) [reply app/views/maps/_facility_auto_complete.js.erb:26]: https://github.com/agilesix/diffusion-marketplace/pull/202/files

File: app/views/maps/_facility_auto_complete.js.erb (hunk 1 above)
Comment (suzonshikdar12) [reply app/views/maps/_facility_auto_complete.js.erb]: ok

Comment (suzonshikdar12) [reply app/views/maps/_facility_auto_complete.js.erb]: I have problem?

**Pull request merged**